src/gatsby-ssr.js

```diff
@@ -1,19 +1,53 @@
+import React from 'react';
 import { ServerStyleSheets } from '@material-ui/styles';
+import postcss from 'postcss';
+import autoprefixer from 'autoprefixer';
+import CleanCSS from 'clean-css';
 
 // Keep track of sheets for each page
 const globalLeak = new Map();
 
+const prefixer = postcss([autoprefixer]);
+const cleanCSS = new CleanCSS();
+
+const defaultOptions = {
+  disableAutoprefixing: false,
+  disableMinification: false,
+};
+
 export const wrapRootElement = ({ element, pathname }, pluginOptions) => {
   const sheets = new ServerStyleSheets(pluginOptions.stylesProvider);
   globalLeak.set(pathname, sheets);
 
   return sheets.collect(element);
 };
 
-export const onRenderBody = ({ setHeadComponents, pathname }) => {
+export const onRenderBody = (
+  { setHeadComponents, pathname },
+  pluginOptions,
+) => {
   const sheets = globalLeak.get(pathname);
+
   if (sheets) {
-    setHeadComponents([sheets.getStyleElement()]);
+    const { disableAutoprefixing, disableMinification } = {
+      ...defaultOptions,
+      ...pluginOptions,
+    };
+
+    let css = sheets.toString();
+
+    css = disableAutoprefixing ? css : prefixer.process(css).css;
```

Comment:
Did you try the changes? I don't think it can work: http://api.postcss.org/Processor.html#process. It returns a promise, you might also want to set `from: undefined` so it uses css as a string source not a filename source.

Reply:
Yes, I tested it. I use the possibility to call postcss (autoprefixer) synchronously. I'm not very happy with that. The docu says:

> Processes input CSS through synchronous plugins, converts Root to a CSS string and returns Result#css.
This property will only work with synchronous plugins. If the processor contains any asynchronous plugins it will throw an error. **This is why this method is only for debug purpose**, you should always use LazyResult#then.

Do you have a better solution?

Reply:
No, I don't know enough.

Reply:
onRenderBody is a sync function and does not wait for an async postcss call